rrmngmnt/foreman_host_wrapper.py

```diff
@@ -0,0 +1,186 @@
+from host import Host
+from foreman.client import Foreman
+
+
+HOSTS_COLLECTION = "hosts"
+
+# Host status constants
+FOREMAN_HOST_STATUS_NO_CHANGES = "No changes"
+FOREMAN_HOST_STATUS_OUT_OF_SYNC = "Out of sync"
+FOREMAN_HOST_STATUS_PENDING_INSTALLATION = "Pending Installation"
+
+
+class ForemanException(Exception):
+    pass
+
+
+class ForemanHost(Host):
+    """
+    Foreman wrapper class
+    """
+
+    def __init__(
+        self, host_ip, foreman_url, foreman_user,
+        foreman_password, api_version=2
+    ):
+        """
+        Initialize ForemanApi class
+
+        :param host_ip: host ip
+        :type host_ip: str
+        :param foreman_url: foreman url
+        :type foreman_url: str
+        :param foreman_user: foreman user
+        :type foreman_user: str
+        :param foreman_password: foreman password
+        :type foreman_password: str
+        :param api_version: foreman api version
+        :type api_version: int
+        """
+        super(ForemanHost, self).__init__(host_ip)
+        self.foreman_api = Foreman(
+            url=foreman_url,
+            auth=(
+                foreman_user, foreman_password
+            ),
+            api_version=api_version
+        )
+        self.host_id = self.get_element_id(self.fqdn, HOSTS_COLLECTION)
+
+    def get_element_id(
+        self, element_name, collection_name, search_attr="name"
+    ):
+        """
+        Get element id from foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :param collection_name: foreman collection name
+        :type collection_name: str
+        :param search_attr: search by attr
+        :type search_attr: str
+        :return: element id
+        :rtype: str
+        """
```

Comment:
more simple would be:

hosts = self.foreman_api..hosts
try:
    return hosts.index(search=elenemt_name)['results'][0]['id']
except IndexError:
    return ""